src/coordinator/shard_map.hpp

```diff
@@ -117,10 +118,10 @@ struct ShardMap {
   Hlc shard_map_version;
   uint64_t max_property_id{kNotExistingId};
   uint64_t max_edge_type_id{kNotExistingId};
-  std::map<PropertyName, PropertyId> properties;
-  std::map<EdgeTypeName, EdgeTypeId> edge_types;
+  storage::v3::NameIdMapper properties;
+  storage::v3::NameIdMapper edge_types;
+  storage::v3::NameIdMapper labels;
```

Comment:
I think keeping the ShardMap is as simple as possible is a good thing, because this structure is used to communicate the information. However by introducing `NameIdMapper` here we are kind of mixing responsibilities of different layers. We already had some not-the-best deicisions (e.g.: the ShardMap should be used to communicate the information only, not to fetch shards/mappings/etc) and with this change we just deepened this mistake.

However it is possible this is only my opinion, so if it is okay for the others, then it is also okay for me.